Function baread() from bacio is called from src/getgb2r.f, but bacio is neither found nor linked...

In subroutine GETGB2R() there are calls to the bacio library:

For example note the calls to BAREAD() around line 206:

```
C  READ AND UNPACK BIT_MAP, IF PRESENT
      IF ( gfld%ibmap.eq.0.OR.gfld%ibmap.eq.254 ) THEN
         ISKIP=LSKIP+SKIP6
         CALL BAREAD(LUGB,ISKIP,4,LREAD,CSIZE)    ! GET LENGTH OF SECTION
         CALL G2_GBYTEC(CSIZE,ILEN,0,32)
         ALLOCATE(CTEMP(ILEN))
         CALL BAREAD(LUGB,ISKIP,ILEN,LREAD,CTEMP)  ! READ IN SECTION
```

This seems to indicate that this subroutine is never used anywhere. If so, we should drop it...

@BoiVuong-NOAA and @Hang-Lei-NOAA do you know anything about this? Was there a past dependency on bacio? Is the subroutine GETGB2R() expected to work by anyone?

That does seem to be incorrect behavior

Went through some old emails to find history.

`find_package(bacio)` was added and then @BoiVuong-NOAA had some issues building and told me g2 didn't require g2.

```
CMake Error at src/CMakeLists.txt:39 (add_library):
  Target "g2_d" links to target "bacio::bacio_4" but the target was not                          ----> G2 library does not require BACIO lirbary.  Why do we tell cmake to find BACIO ?
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```

Interestingly, though, we've had g2 without bacio for over a year, and it has been installed on WCOSS2 with everyone testing their applications for production, and it hasn't come up...

--- a/.github/workflows/Intel.yml
+++ b/.github/workflows/Intel.yml
@@ -32,6 +32,21 @@ jobs:
         sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-mpi-devel intel-oneapi-openmp intel-oneapi-compiler-fortran intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
         echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
       
+    - name: checkout-bacio
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-bacio
+        path: bacio
+        ref: develop
+
+    - name: build-bacio
+      run: |
+        cd bacio
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -65,7 +80,7 @@ jobs:
         cd g2
         mkdir build 
         cd build
-        cmake .. -DOPENMP=ON -DJasper_ROOT=~/Jasper
+        cmake .. -DOPENMP=ON -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio
         make -j2 VERBOSE=1
     
     - name: test

--- a/.github/workflows/gcc_docs_test-coverage.yml
+++ b/.github/workflows/gcc_docs_test-coverage.yml
@@ -17,6 +17,21 @@ jobs:
         sudo apt-get install libpng-dev zlib1g-dev libjpeg-dev doxygen
         python3 -m pip install gcovr
 
+    - name: checkout-bacio
+      uses: actions/checkout@v2
+      with:
+        repository: NOAA-EMC/NCEPLIBS-bacio
+        path: bacio
+        ref: develop
+
+    - name: build-bacio
+      run: |
+        cd bacio
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
+        make -j2
+        make install
     - name: checkout-jasper
       uses: actions/checkout@v2
       with:
@@ -52,7 +67,7 @@ jobs:
         mkdir build
         doxygen --version
         cd build
-        cmake .. -DJasper_ROOT=~/Jasper -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address"
+        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address"
         make -j2 VERBOSE=1 
 
     - name: test

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,6 +23,7 @@ endif()
 
 find_package(Jasper REQUIRED)
 find_package(PNG REQUIRED)
+find_package(bacio REQUIRED)
 
 add_subdirectory(src)
 

--- a/cmake/PackageConfig.cmake.in
+++ b/cmake/PackageConfig.cmake.in
@@ -12,7 +12,6 @@ include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
 include(CMakeFindDependencyMacro)
 
 find_dependency(PNG)
-find_dependency(Jasper)
 find_dependency(bacio)
 
 get_target_property(@PROJECT_NAME@_BUILD_TYPES @PROJECT_NAME@::@PROJECT_NAME@_4 IMPORTED_CONFIGURATIONS)

--- a/cmake/PackageConfig.cmake.in
+++ b/cmake/PackageConfig.cmake.in
@@ -12,6 +12,8 @@ include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
 include(CMakeFindDependencyMacro)
 
 find_dependency(PNG)
+find_dependency(Jasper)
+find_dependency(bacio)
 
 get_target_property(@PROJECT_NAME@_BUILD_TYPES @PROJECT_NAME@::@PROJECT_NAME@_4 IMPORTED_CONFIGURATIONS)
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -67,7 +67,7 @@ foreach(kind ${kinds})
   target_link_libraries(${lib_name}_c PRIVATE PNG::PNG)
 
   add_library(${lib_name} STATIC $<TARGET_OBJECTS:${lib_name}_f> $<TARGET_OBJECTS:${lib_name}_c>)
-  target_link_libraries(${lib_name} PUBLIC PNG::PNG ${JASPER_LIBRARIES})
+  target_link_libraries(${lib_name} PUBLIC bacio::bacio_4 PNG::PNG ${JASPER_LIBRARIES})
   target_include_directories(${lib_name} INTERFACE
     $<BUILD_INTERFACE:${module_dir}>
     $<INSTALL_INTERFACE:include_${kind}>)

